app/models/activity.rb

```diff
@@ -211,13 +211,11 @@ class Activity < ApplicationRecord
   scope :with_roda_identifier, -> { where.not(roda_identifier_compound: nil) }
 
   scope :projects_and_third_party_projects_for_report, ->(report) {
-    programmes = where(level: :programme, parent_id: report.fund_id)
-    projects = where(level: :project, parent_id: programmes.pluck(:id))
-    third_party_projects = where(level: :third_party_project, parent_id: projects.pluck(:id))
-
-    for_organisation = where(organisation_id: report.organisation_id)
-
-    for_organisation.merge(projects.or(third_party_projects))
+    where(
+      level: [:project, :third_party_project],
+      organisation_id: report.organisation_id,
+      source_fund_code: report.fund.source_fund_code
+    )
   }
 
```

Comment:
This is so much easer to understand (and no doubt faster too).

I feel like the ideal presentation of the changes would be to have a single commit where you perform this refactoring of `Activity:projects_and_third_party_projects_for_report` and the matching spec **remains unchanged**. That would make it super clear that there is no change in behaviour, just in implementation. I'm guessing that you might need an earlier commit in which you refactor the spec so it supports the new implementation, preemptively?

Reply:
Good call - I've done that now. Makes it much easier to follow I think